### PR TITLE
Remove entire compatibility layer and updates id generation

### PR DESCRIPTION
**Pull request by tylersmalley**
When creating an id, we will always prefix the type to either an id which was provided or one we generate. This allows us to simplify all actions for managing documents.

### CODE REVIEW

File: src/server/saved_objects/client/saved_objects_client.js [hunk 1]
```diff
@@ -216,27 +213,17 @@ export class SavedObjectsClient {
...
-    const [hit] = get(docResponse, 'hits.hits', []);
-
-    if (!hit) {
-      throw Boom.notFound();
-    }
-
     const response = await this._withKibanaIndex('update', {
-      id: hit._id,
       type: V6_TYPE,
+      id: `${type}:${id}`,
```
Comment (spalger) [inline src/server/saved_objects/client/saved_objects_client.js]: This should probably use `this._generateEsId()`

File: src/server/saved_objects/client/saved_objects_client.js [hunk 2]
```diff
@@ -249,4 +236,8 @@ export class SavedObjectsClient {
       throw handleEsError(err);
     }
   }
+
+  generateEsId(type, id) {
```
Comment (spalger) [inline src/server/saved_objects/client/saved_objects_client.js]: This should probably be `_` prefixed, it's really an implementation detail and shouldn't be a part of the "public" API

**Review by spalger — CHANGES_REQUESTED** [review]

File: src/server/saved_objects/client/saved_objects_client.js (hunk 1 above)
Comment (tylersmalley) [reply src/server/saved_objects/client/saved_objects_client.js]: 👍

File: src/server/saved_objects/client/saved_objects_client.js (hunk 2 above)
Comment (tylersmalley) [reply src/server/saved_objects/client/saved_objects_client.js]: 👍

**Pull request merged**